include/LUSolver.hpp

```diff
@@ -12,27 +12,23 @@
 
 namespace blitzdg {
   class LUSolver {
-      index_type N;
-      matrix_type* A;
+      const matrix_type* A;
 
       // Umfpack-specific fields
       index_type * Ap;
       index_type * Ai;
       real_type * Ax;
       index_type * Map;
-      real_type * null;
 
       void * Symbolic;
       void * Numeric;
-
-      SparseTriplet Triplet;
-
+      
       SparseMatrixConverter MatrixConverter;
     
     public:
-      LUSolver(matrix_type* const &, SparseMatrixConverter const &);
+      explicit LUSolver(const matrix_type* Ain);
```

Comment:
What kind of implicit conversion are we preventing by prefixing the constructor with explicit here?

Reply:
Any time you have a constructor that takes only a single argument it is generally good practice to prefix such a constructor with the explicit keyword. In this case, we avoid the implicit conversion of a pointer to matrix_type to an LUSolver.